ir/instr.cpp

```diff
@@ -227,6 +231,14 @@ StateValue BinOp::toSMT(State &s) const {
     not_poison &= (b == 0u || a != 0u);
     break;
 
+  case SAdd_Overflow:
+    val = a + b;
+    val = val.concat(expr::mkIf(a.add_no_soverflow(b), expr::mkUInt(0, 1), expr::mkUInt(1, 1)));
```

Comment:
it may not matter but it seems simpler to use !add_no_soverflow() instead of an ITE

Reply:
that doesn't work. add_no_soverflow returns a true/false expr and implicit cast to i1 doesn't work.